vita3k/gui/src/archive_install_dialog.cpp

```diff
@@ -34,25 +36,59 @@ void draw_archive_install_dialog(GuiState &gui, HostState &host) {
 
     static bool draw_file_dialog = true;
     static bool content_install_confirm = false;
+    static bool finished_installing = false;
+    static float progress;
+    static bool reinstalling;
 
     if (draw_file_dialog) {
         result = NFD_OpenDialog("vpk,zip", nullptr, &archive_path);
         draw_file_dialog = false;
+        finished_installing = false;
 
         if (result == NFD_OKAY) {
-            if (install_archive(host, &gui, fs::path(string_utils::utf_to_wide(archive_path))))
-                content_install_confirm = true;
-            else if (!gui.content_reinstall_confirm)
-                ImGui::OpenPopup("Content installation failed");
+            std::thread installation([&host, &gui]() {
+                if (install_archive(host, &gui, fs::path(string_utils::utf_to_wide(archive_path)), &progress))
+                    content_install_confirm = true;
+                else if (!gui.content_reinstall_confirm)
+                    ImGui::OpenPopup("Content installation failed");
+                finished_installing = true;
```

Comment:
It's little dirty to use flag (finished_installing) here (because it has race condition) Try to use condition_variable.